### PR TITLE
Update Contract Info with RoundingIntervals

### PR DESCRIPTION
**Pull request by nkohen**
Adds RoundingIntervals to MultiDigitContractInfo completing integration with the rest of the DLC code.

Adds RoundingIntervals to existing multi-digit unit tests.

### CODE REVIEW

File: app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCMessage.scala [hunk 1]
```diff
@@ -564,15 +575,19 @@ object DLCMessage {
...
-                changeAddress = changeAddress,
-                cetSigs = cetSigs,
-                tempContractId = tempContractId)
+      DLCAccept(
+        totalCollateral = totalCollateral,
+        pubKeys = pubKeys,
+        fundingInputs = fundingInputs,
+        changeAddress = changeAddress,
+        cetSigs = cetSigs,
+        roundingIntervalsOpt,
```
Comment (benthecarman) [inline app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCMessage.scala]: Nit : add variable name

File: app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCMessage.scala [hunk 2]
```diff
@@ -594,8 +610,9 @@ object DLCMessage {
         fundingInputs = fundingInputs.map(_.toTLV),
         changeSPK = changeAddress.scriptPubKey,
         cetSignatures = CETSignaturesV0TLV(cetSigs.adaptorSigs),
-        refundSignature =
-          ECDigitalSignature.fromFrontOfBytes(cetSigs.refundSig.signature.bytes)
+        refundSignature = ECDigitalSignature.fromFrontOfBytes(
+          cetSigs.refundSig.signature.bytes),
+        roundingIntervalsOpt.getOrElse(RoundingIntervals.noRounding).toTLV
```
Comment (benthecarman) [inline app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCMessage.scala]: Ditto

File: app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/OutcomeValueFunction.scala [hunk 3]
```diff
@@ -12,6 +13,12 @@ case class OutcomeValueFunction(points: Vector[OutcomeValuePoint]) {
           },
           s"Points must be ascending: $points")
 
+  def toTLV: PayoutFunctionV0TLV = {
```
Comment (benthecarman) [inline app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/OutcomeValueFunction.scala]: Lazy val

File: app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/OutcomeValueFunction.scala [hunk 4]
```diff
@@ -12,6 +13,12 @@ case class OutcomeValueFunction(points: Vector[OutcomeValuePoint]) {
...
+    PayoutFunctionV0TLV(points.map { point =>
+      TLVPoint(point.outcome.toLongExact, point.value, point.isEndpoint)
```
Comment (benthecarman) [inline app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/OutcomeValueFunction.scala]: Won't toLongExact cause an exception if it's a decimal here?

File: app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/RoundingIntervals.scala [hunk 5]
```diff
@@ -19,6 +20,12 @@ case class RoundingIntervals(intervalStarts: Vector[(BigDecimal, Long)]) {
             s"Intervals must be ascending: $intervalStarts")
   }
 
+  def toTLV: RoundingIntervalsV0TLV = {
+    RoundingIntervalsV0TLV(intervalStarts.map {
+      case (outcome, amt) => (outcome.toLongExact, Satoshis(amt))
```
Comment (benthecarman) [inline app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/RoundingIntervals.scala]: Double ditto

File: core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala [hunk 6]
```diff
@@ -778,6 +794,42 @@ object ContractInfoV0TLV extends TLVFactory[ContractInfoV0TLV] {
   }
 }
 
+case class RoundingIntervalsV0TLV(intervalStarts: Vector[(Long, Satoshis)])
+    extends TLV {
+  def isEmpty: Boolean = intervalStarts.isEmpty
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala:860]: Val

Comment (benthecarman) [reply core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala:860]: Maybe instead separate this into a separate type

File: core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala [hunk 7]
```diff
@@ -1132,11 +1204,18 @@ case class DLCAcceptTLV(
...
     cetSignatures: CETSignaturesTLV,
-    refundSignature: ECDigitalSignature)
+    refundSignature: ECDigitalSignature,
+    roundingIntervalsV0TLV: RoundingIntervalsV0TLV =
+      RoundingIntervalsV0TLV.noRounding)
     extends TLV {
   override val tpe: BigSizeUInt = DLCAcceptTLV.tpe
 
   override val value: ByteVector = {
+    // TODO: Use of rounding intervals option here is a hack
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala]: Ping

**Comment by benthecarman** [comment]
Looks like DLC tests are failing

**Comment by nkohen** [comment]
> Looks like DLC tests are failing

I believe that's just test vectors due to breaking changes, nothing concerning

File: app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/OutcomeValueFunction.scala (hunk 4 above)
Comment (nkohen) [reply app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/OutcomeValueFunction.scala]: It will and should

**Comment by benthecarman** [comment]
You should be able to set the rounding interval in the preview graph now:

https://github.com/bitcoin-s/bitcoin-s/blob/adaptor-dlc/app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/ViewDLCDialog.scala#L200

File: core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCPayoutCurve.scala [hunk 8]
```diff
@@ -12,6 +13,15 @@ case class DLCPayoutCurve(points: Vector[OutcomePayoutPoint]) {
           },
           s"Points must be ascending: $points")
 
+  def toTLV: PayoutFunctionV0TLV = {
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCPayoutCurve.scala:16]: nit: lazy val

File: core/src/main/scala/org/bitcoins/core/protocol/dlc/RoundingIntervals.scala [hunk 9]
```diff
@@ -23,6 +24,12 @@ case class RoundingIntervals(intervalStarts: Vector[IntervalStart]) {
             s"Intervals must be ascending: $intervalStarts")
   }
 
+  def toTLV: RoundingIntervalsV0TLV = {
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/protocol/dlc/RoundingIntervals.scala:27]: ditto

File: core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala [hunk 10]
```diff
@@ -1172,6 +1240,55 @@ object DLCOfferTLV extends TLVFactory[DLCOfferTLV] {
...
+object NoNegotiationFieldsTLVFactory
+    extends TLVFactory[NoNegotiationFieldsTLV.type] {
+  override val tpe: BigSizeUInt = BigSizeUInt(55334)
+
+  override def fromTLVValue(value: ByteVector): NoNegotiationFieldsTLV.type = {
+    require(value.isEmpty, "NoNegotiationsFieldsTLV must be empty")
+
+    NoNegotiationFieldsTLV
+  }
+}
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala:1268]: nit: imo `EmptyNegotiationFieldsTLV` would be a better name, but I am indifferent

**Review by benthecarman — APPROVED** [review]

**Pull request merged**